trin-core/src/portalnet/overlay.rs

```diff
@@ -490,6 +675,24 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
     }
 }
 
+/// Randomly select log2 nodes. log2() is stable only for floats, that's why we cast it first to f32
+fn log2_random_enrs(enrs: Vec<Enr>) -> Vec<Enr> {
+    assert!(!enrs.is_empty());
```

Comment:
If there is a panic here, then we should document that in doc comment. I don't think an empty input should be a panic. We can return `Option<Vec<Enr>>` instead. Alternatively, we can return the input if the input is empty.

Reply:
I think returning an `Error` and safe `unwrap` also makes sense?

Reply:
I prefer `Option` where `None` represents "undefined", but I also see that `Error` could make sense. We could use something like `unwrap_or_else` and return/continue on `None` or `Err`.